mlir/lib/Dialect/LLVMIR/IR/LLVMTypes.cpp

```diff
@@ -150,8 +150,7 @@ generatedTypePrinter(Type def, AsmPrinter &printer);
 
 bool LLVMArrayType::isValidElementType(Type type) {
   return !llvm::isa<LLVMVoidType, LLVMLabelType, LLVMMetadataType,
-                    LLVMFunctionType, LLVMTokenType, LLVMScalableVectorType>(
-      type);
+                    LLVMFunctionType, LLVMTokenType>(type);
```

Comment:
Note: `LLVMScalableVectorType` was in here, but I think it should not have been because LLVM does not support it.

```c++
bool ArrayType::isValidElementType(Type *ElemTy) {
  return !ElemTy->isVoidTy() && !ElemTy->isLabelTy() &&
         !ElemTy->isMetadataTy() && !ElemTy->isFunctionTy() &&
         !ElemTy->isTokenTy() && !ElemTy->isX86_AMXTy();
}
```

Reply:
That makes sense but I don't remember if we did something about this for SME. cc: @banach-space

Reply:
That `LLVMScalableVectorType` was added long before SME:
* https://reviews.llvm.org/D85663

But yes, "arrays of scalable vectors" are a thing and we rely on them. That said, I don't see any SME/SVE tests failing (I also check e2e locally), so this should be fine.